app/src/renderer/vuex/modules/wallet.js

```diff
@@ -65,10 +73,37 @@ export default ({ commit, node }) => {
       if (!res) return
       commit('setWalletSequence', res.data)
     },
-    async queryWalletHistory ({ state, commit }) {
+    async queryWalletHistory ({ state, commit, dispatch }) {
       let res = await node.coinTxs(state.key.address)
       if (!res) return
       commit('setWalletHistory', res)
+      let blockHeights = new Set()
+      res.forEach(t => blockHeights.add(t.height))
+      blockHeights.forEach(h => {
+        dispatch('queryTransactionTime', h)
+      })
+    },
+    async queryTransactionTime ({ commit, dispatch }, blockHeight) {
+      let block = await dispatch('queryBlock', blockHeight)
+      commit('setTransactionTime', { blockHeight, block })
+    },
+    async queryBlock ({ state, commit }, height) {
+      let block = state.blocks.find(b => b.block_meta.header.height === height)
+      if (block) {
+        return block
+      }
+      block = await new Promise((resolve, reject) => {
+        node.rpc.block({height}, (err, block) => {
```

Comment:
We should probably just get the block header, instead of the whole thing (the difference is that the block contains all the transaction data, which may grow to be large in the future).

We can get only the header by using `node.rpc.blockchain({ minHeight: height, maxHeight: height }, cb)` (looks a little less clean, but probably worth it). The result format is similar but contains `block_metas` as an array.